examples/fifaText/main.go

```diff
@@ -0,0 +1,317 @@
+package main
+
+import (
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sjwhitworth/golearn/evaluation"
+	"github.com/sjwhitworth/golearn/trees"
+
+	"github.com/kniren/gota/dataframe"
+	"github.com/sjwhitworth/golearn/base"
+	"gonum.org/v1/plot"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+)
+
+var (
+	command *string
+)
+
+// WordData is data that will help with training a model that detects names.
+type WordData struct {
+	Word       string
+	Occurances int
+	// Vector     word2vec.Vector
+	IsName bool
+}
+
+// Tweet object for holding raw form of csv data.
+type Tweet struct {
+	ID            int
+	Lang          string
+	Date          int64
+	Source        string
+	Length        int
+	OrgTweet      string
+	Tweets        string
+	Likes         int
+	Retweets      int
+	Hashtag       string
+	Usermention   string
+	UserMentionID string
+	Name          string
+	Place         string
+	Followers     int
+	Friends       int
+}
+
+func init() {
+	command = flag.String("action", "train", "This is the method that the program in performing.")
+}
+func main() {
+	flag.Parse()
+	fmt.Println(*command)
+	file := "../datasets/FIFA.csv"
+	switch *command {
+	case "plot":
+		err := UnderstandData(file)
+		log.Println(err)
+	case "train":
+		err := DecisionTree(("data/words.csv"))
+		log.Println((err))
+	default:
+
+	}
+}
+
+// DecisionTree Write and train DecisionTree model.
+func DecisionTree(file string) error {
+	// var tree base.Classifier
+
+	rand.Seed(44111342)
+
+	// Load in the words dataset
+	words, err := base.ParseCSVToInstances(file, true)
+	if err != nil {
+		return err
+	}
+
+	// Create a 70-30 training-test split
+	trainData, testData := base.InstancesTrainTestSplit(words, 0.70)
+
+	//
+	// First up, use ID3
+	//
+	tree := trees.NewID3DecisionTree(0.6)
+	// (Parameter controls train-prune split.)
+
+	// Train the ID3 tree
+	err = tree.Fit(trainData)
+	if err != nil {
+		return err
+	}
+
+	// Generate predictions
+	predictions, err := tree.Predict(testData)
+	if err != nil {
+		return err
+	}
+
+	// Evaluate
+	fmt.Println("ID3 Performance (information gain)")
+	cf, err := evaluation.GetConfusionMatrix(testData, predictions)
+	if err != nil {
+		return fmt.Errorf("Unable to get confusion matrix: %s", err.Error())
+	}
+	fmt.Println(evaluation.GetSummary(cf))
+
+	tree.Save("models/DecisonTree.h")
+
+	// Next up, Random Trees
+
+	// Consider two randomly-chosen attributes
+	tree2 := trees.NewRandomTree(2)
+	err = tree2.Fit(trainData)
+	if err != nil {
+		panic(err)
```

Comment:
Why does this code panic here instead of returning the error?

Reply:
thanks... some of the code was copied and not cleaned